plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergTableHandle.java

```diff
@@ -92,6 +115,19 @@ public TableType getTableType()
         return enforcedPredicate;
     }
 
+    // Not available on workers
+    @JsonIgnore
+    public Optional<TableMetadata> getIcebergTableMetadata()
+    {
+        return icebergTableMetadata;
+    }
+
+    public TableMetadata doGetIcebergTableMetadata()
```

Comment:
The `do` prefix implies an action. I would call this `getRequiredIcebergTableMetadata()` or `getIcebergTableMetadataOrElseThrow()`.